x-pack/plugin/core/src/main/java/org/elasticsearch/xpack/core/ilm/SearchableSnapshotAction.java

```diff
@@ -0,0 +1,203 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+package org.elasticsearch.xpack.core.ilm;
+
+import org.elasticsearch.action.ActionListener;
+import org.elasticsearch.action.admin.cluster.snapshots.status.SnapshotStatus;
+import org.elasticsearch.action.admin.cluster.snapshots.status.SnapshotsStatusRequest;
+import org.elasticsearch.action.admin.cluster.snapshots.status.SnapshotsStatusResponse;
+import org.elasticsearch.client.Client;
+import org.elasticsearch.cluster.SnapshotsInProgress;
+import org.elasticsearch.cluster.metadata.IndexMetaData;
+import org.elasticsearch.common.ParseField;
+import org.elasticsearch.common.Strings;
+import org.elasticsearch.common.TriConsumer;
+import org.elasticsearch.common.io.stream.StreamInput;
+import org.elasticsearch.common.io.stream.StreamOutput;
+import org.elasticsearch.common.xcontent.ConstructingObjectParser;
+import org.elasticsearch.common.xcontent.XContentBuilder;
+import org.elasticsearch.common.xcontent.XContentParser;
+import org.elasticsearch.snapshots.SnapshotException;
+import org.elasticsearch.xpack.core.ilm.Step.StepKey;
+
+import java.io.IOException;
+import java.util.Arrays;
+import java.util.List;
+import java.util.Objects;
+
+import static org.elasticsearch.cluster.SnapshotsInProgress.State.ABORTED;
+import static org.elasticsearch.cluster.SnapshotsInProgress.State.FAILED;
+import static org.elasticsearch.cluster.SnapshotsInProgress.State.MISSING;
+import static org.elasticsearch.cluster.SnapshotsInProgress.State.SUCCESS;
+
+/**
+ * A {@link LifecycleAction} that will convert the index into a searchable snapshot, by taking a snapshot of the index, creating a
+ * searchable snapshot and the corresponding "searchable snapshot index", deleting the original index and swapping its aliases to the
+ * newly created searchable snapshot backed index.
+ */
+public class SearchableSnapshotAction implements LifecycleAction {
+    public static final String NAME = "searchable_snapshot";
+
+    public static final ParseField SNAPSHOT_REPOSITORY = new ParseField("snapshot_repository");
+    public static final ParseField SEARCHABLE_REPOSITORY = new ParseField("searchable_repository");
+
+    public static final String RESTORED_INDEX_PREFIX = "ilm-restored-";
```

Comment:
We might want to discuss what we want the prefix to be, I'm not sure about "ilm-restored-" in case we ever have a different (non-searchable) ILM action to restore a snapshot, what do you think?

Reply:
Hm, I'm not sure I understand what you mean. The intention was to keep it "searchable snapshot" independent, which it is now. Do you have something specific in mind? (and this is also configurable in the `RestoreSnapshotStep`, in case we reuse it with regulard snapshot restores)

Reply:
What about `restored-`? I think that fits a little more with our `shrink-` prefix